File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 2/3_greatest_common_divisor/gcd.py
import sys

def EuclidGCD(a, b):
    if b == 0:
        return a
    
    rem = a % b
    return EuclidGCD(b, rem)

if __name__ == '__main__':
    a, b = input().split()
    print(EuclidGCD(int(a), int(b)))

File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 2/3_greatest_common_divisor/gcd.cpp
#include <iostream>
using namespace std;

int EuclidGCD(int a, int b){
    if (b == 0){
        return a;
    }
    int rem = a % b;
    return EuclidGCD(b, rem);
}

int main(){
    int a, b;
    cin >> a >> b;
    cout << EuclidGCD(a, b);
    return 0;
}